src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputList) {
+        inputList.remove(0);
+        return inputList.stream()
+                 .map(s -> s.split(","))
+                 .map(s -> new FruitTransaction(FruitTransaction.Operation.getFromChar(s[0]),
+                                                s[1], Integer.parseInt(s[2])))
+                 .toList();

```

Comment:
There's no validation to ensure that the split array 's' has the expected number of elements or that the elements can be parsed correctly. You should add checks to handle cases where the input data doesn't have the correct format.